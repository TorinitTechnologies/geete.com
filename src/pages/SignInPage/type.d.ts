declare interface SignInFields {
	email: string | null;
	password: string | null;
}
