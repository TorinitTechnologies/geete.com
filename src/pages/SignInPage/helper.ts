import * as Yup from 'yup';

export const getLoginSchema = () => {
	const schema = Yup.object({
		email: Yup.string()
			.email('Please enter valid email')
			.required('Email id is required'),
		password: Yup.string().required('Password is required'),
	});
	return schema;
};
