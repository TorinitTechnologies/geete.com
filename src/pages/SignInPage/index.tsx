import { useFormik } from 'formik';
import { useNavigate } from 'react-router-dom';

import Button from '@atoms/Button';
import Input from '@atoms/Input';

import { makeUserLoggedIn } from '@utils/authUtils/index';
import { getLoginSchema } from './helper';

function SignInPage() {
	const navigate = useNavigate();
	const onLogin = () => {
		makeUserLoggedIn();
		navigate('/');
	};

	const loginSchema = getLoginSchema();

	const formik = useFormik({
		validationSchema: loginSchema,
		initialValues: { email: null, password: null },
		onSubmit: onLogin,
	});

	const { errors, handleChange, touched, handleSubmit } = formik;

	return (
		<div data-testid={'sign_in_page'}>
			<Input
				id={'email'}
				name={'email'}
				label={'Enter Email'}
				dataTestId={'email'}
				variant={'outlined'}
				errorDataTestId={'email_error'}
				error={(touched.email && errors.email) || ''}
				onChange={handleChange}
			/>
			<Input
				id={'password'}
				name={'password'}
				label={'Enter Password'}
				dataTestId={'password'}
				variant={'outlined'}
				type={'password'}
				onChange={handleChange}
				errorDataTestId={'password_error'}
				error={(touched.password && errors.password) || ''}
			/>
			<Button
				id={'login'}
				name={'login'}
				dataTestId={'login_button'}
				variant={'contained'}
				title={'Login'}
				text={'Login'}
				onClick={handleSubmit}
			/>
		</div>
	);
}

export default SignInPage;
