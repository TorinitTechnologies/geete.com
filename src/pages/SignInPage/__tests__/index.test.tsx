const mockedUsedNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
	...jest.requireActual('react-router-dom'),
	useNavigate: () => mockedUsedNavigate,
}));

import { fireEvent, render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';

import SignInPage from '../index';

describe('SignInPage', () => {
	it('renders correctly', () => {
		const tree = renderer.create(<SignInPage />).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('SignInPage should render', async () => {
		render(<SignInPage />);
		const signInPage = await screen.findByTestId('sign_in_page');
		expect(signInPage).toBeInTheDocument();
	});

	it('should have enter email text field', async () => {
		render(<SignInPage />);
		const emailField = await screen.findByTestId('email');
		expect(emailField).toBeInTheDocument();
	});

	it('should have enter password text field', async () => {
		render(<SignInPage />);
		const passwordField = (await screen.findByTestId(
			'password'
		)) as HTMLElement;
		expect(passwordField).toBeInTheDocument();
	});

	it('should have login button', async () => {
		render(<SignInPage />);
		const loginButton = await screen.findByTestId('login_button');
		expect(loginButton).toBeInTheDocument();
	});

	it('should show enter email validation message', async () => {
		render(<SignInPage />);
		const loginButton = await screen.findByTestId('login_button');
		await fireEvent.click(loginButton);
		const errorMessage = await screen.findByTestId('email_error');
		expect(errorMessage).toBeInTheDocument();
	});

	it('should not show enter email validation message', async () => {
		render(<SignInPage />);
		const emailField = await screen.getByTestId('email').querySelector('input');
		const loginButton = await screen.findByTestId('login_button');
		if (emailField) {
			fireEvent.change(emailField, { target: { value: 'test@t.ca' } });
			fireEvent.click(loginButton);
			const errorMessage = await screen.queryByTestId('email_error');
			expect(errorMessage).toBeFalsy();
		}
	});

	it('should show enter password validation message', async () => {
		render(<SignInPage />);
		const loginButton = await screen.findByTestId('login_button');
		await fireEvent.click(loginButton);
		const errorMessage = await screen.findByTestId('password_error');
		expect(errorMessage).toBeInTheDocument();
	});
});
