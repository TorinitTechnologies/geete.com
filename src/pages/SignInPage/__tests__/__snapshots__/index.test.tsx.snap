// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignInPage renders correctly 1`] = `
<div
  className="sc-dkPulx"
  data-testid="sign_in_page"
>
  <main
    className="MuiGrid-root MuiGrid-container css-1jrn2fe-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-4 MuiGrid-grid-md-7 css-1io5csw-MuiGrid-root"
    />
    <div
      className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation6 MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-8 MuiGrid-grid-md-5 css-gadse4-MuiPaper-root-MuiGrid-root"
    >
      <div
        className="MuiBox-root css-1y8ugea"
      >
        <div
          className="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-1999axt-MuiAvatar-root"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="LockOutlinedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
            />
          </svg>
        </div>
        <h1
          className="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
        >
          Sign in
        </h1>
        <form
          className="MuiBox-root css-164r41r"
          noValidate={true}
          onSubmit={[Function]}
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <div
            className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root sc-gsDKeI css-wb57ya-MuiFormControl-root-MuiTextField-root"
            data-testid="email"
            style={
              Object {
                "marginTop": 30,
              }
            }
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink={false}
              htmlFor="email"
              id="email-label"
            >
              Enter Email
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                disabled={false}
                id="email"
                name="email"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="text"
              />
              <fieldset
                aria-hidden={true}
                className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="css-yjsfm1"
                >
                  <span>
                    Enter Email
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root sc-gsDKeI css-wb57ya-MuiFormControl-root-MuiTextField-root"
            data-testid="password"
            style={
              Object {
                "marginTop": 30,
              }
            }
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink={false}
              htmlFor="password"
              id="password-label"
            >
              Enter Password
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                disabled={false}
                id="password"
                name="password"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="password"
              />
              <fieldset
                aria-hidden={true}
                className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="css-yjsfm1"
                >
                  <span>
                    Enter Password
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth sc-bdvumH css-1vhaqj4-MuiButtonBase-root-MuiButton-root"
            data-testid="login_button"
            disabled={false}
            id="login"
            name="login"
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            style={
              Object {
                "marginTop": 50,
              }
            }
            tabIndex={0}
            title="Login"
            type="button"
          >
            Login
          </button>
        </form>
      </div>
    </div>
  </main>
</div>
`;
